tests/IssueKeyResolverTest.php

```diff
@@ -34,6 +35,19 @@ public function testIssueKeyResolved($expectedIssueKey, $branchName)
     }
 
     /**
+     * @param string $issueKey
+     * @param string $branchName
+     *
+     * @dataProvider dataProvider
+     */
+    public function testIssueKeyResolvedWithProject($expectedIssueKey, $branchName)
```

Comment:
In that test you alter the branch name that is tested. It's not clear what you are trying to achieve. Maybe the following would make it more explicit :

```
$resolver = new IssueKeyResolver();
$branchNameWithoutProjectKey = str_replace('FOO-', '', $branchName);
$projectKey = 'FOO';
$actualIssueKey = $resolver->resolveKeyFromBranchName($branchNameWithoutProjectKey, $projectKey);
$this->assertEquals($expectedIssueKey, $actualIssueKey);
```